src/Test/All.hs

```diff
@@ -8,6 +9,10 @@ import Data.List
 import System.Directory
 import System.FilePath
 
+#if __GLASGOW_HASKELL__ < 710
+import Data.Functor ((<$>))
+#endif
```

Comment:
Yes, both versions suck, but my opinion is that the version with CPP sucks more - with CPP there is code that is out of the reach of static checkers, lint systems etc, and that adds significant overhead running the preprocessor (on certain systems it can be on the order of 1s). I suspect the processor test would have to persist for ~7 years, so that's quite a long way in the future, but when I do drop GHC 7.8 support I'll probably go through and clean it up (or hopefully write a lint check to do so for everyone).

And I entirely agree, the way around it is total magic.